oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/importer/IndexImporter.java

```diff
@@ -327,4 +408,25 @@ public String toString() {
         }
     }
 
+    interface IndexImporterStepExecutor {
+        void execute() throws CommitFailedException, IOException;
+    }
+
+    void runWithRetry(int maxRetries, IndexImportState indexImportState, IndexImporterStepExecutor step) throws CommitFailedException, IOException {
+        int count = 0;
+        while (count < maxRetries) {
+            log.info("IndexImporterStepExecutor:{} ,count:{}", indexImportState, count);
+            try {
+                step.execute();
+                break;
+            } catch (CommitFailedException | IOException e) {
+                log.info("IndexImporterStepExecutor:{} fail count: {}", indexImportState, count);
```

Comment:
I would log.warn here, and also log the exception stack trace (that is, append ", e").